### PR TITLE
Feature/notification

### PR DESCRIPTION
**Pull request by Laroche-Nicolas**

### CODE REVIEW

**Review by MrZyr0 — APPROVED** [review]

File: src/components/Menu/Menu.vue [hunk 1]
```diff
@@ -1,66 +1,157 @@
...
+    <div class="menu__container">
+      <main>
+        <span v-if="notificationAlertMsg">{{ notificationAlertMsg }}</span>
+        <h1>Battle Quiche</h1>
+        <h3 v-if="!!user">Bonjour {{ user.username }} !</h3>
+      </main>
+      <aside>
+        <button @click="createParty">Créer une partie</button>
+        <!--       TODO: Fix CreateParty route parms to fix this router-link-->
+        <!--      <router-link :to="{ name: 'CreateParty' }">Créer une partie</router-link>-->
```
Comment (MrZyr0) [inline src/components/Menu/Menu.vue]:
```suggestion
        <router-link :to="{ name: 'CreateParty' }">Créer une partie</router-link>
```

**Review by MrZyr0 — CHANGES_REQUESTED** [review]

**Review by zaktolba — APPROVED** [review]

**Review by MrZyr0 — APPROVED** [review]

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -3,6 +3,7 @@ on:
   push:
     branches:
       - main
+      - feature/notification
```
Comment (MaxenceMottard) [inline .github/workflows/deploy.yml]: remove

File: public/sw.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+self.addEventListener("push", (event) => {
+    let data = {
+        title: 'Empty notification',
+        message: 'new empty notification receive'
+    };
+
+    // FIXME: event data on manifest test
+    // eslint-disable-next-line no-prototype-builtins
+    // if ( event.hasOwnProperty('data') && event.data && Object.keys(event.data).length > 0 ) { // notification has data
```
Comment (MaxenceMottard) [inline public/sw.js:13]: maybe remove comments

File: public/sw.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    //     console.log('Notification data parsed !');
+        data = event.data.json();
+    // }
+
+    event.waitUntil(self.registration.showNotification(data.title, {
+        body: data.message,
+        data
+    }));
+
+    console.log(event);
```
Comment (MaxenceMottard) [inline public/sw.js]: remove

File: src/components/Menu/Menu.vue [hunk 5]
```diff
@@ -1,66 +1,157 @@
...
   },
   methods: {
     ...mapActions(['setUser']),
     async setupUsername(e) {
       e.preventDefault();
 
       try {
-        const URL_USR = `${URL}/user`
-        const result = await this.$http.put(URL_USR, {username : this.username})
+        const URL_USR = `${process.env.VUE_APP_API_BASE_URL}/user`
```
Comment (MaxenceMottard) [inline src/components/Menu/Menu.vue]: maybe use this.$env

File: src/components/Menu/Menu.vue [hunk 6]
```diff
@@ -1,66 +1,157 @@
...
+    <div class="menu__container">
+      <main>
+        <span v-if="notificationAlertMsg">{{ notificationAlertMsg }}</span>
+        <h1>Battle Quiche</h1>
+        <h3 v-if="!!user">Bonjour {{ user.username }} !</h3>
+      </main>
+      <aside>
+        <button @click="createParty">Créer une partie</button>
+        <!--       TODO: Fix CreateParty route parms to fix this router-link-->
+        <router-link :to="{ name: 'CreateParty' }">Créer une partie</router-link>
```
Comment (MaxenceMottard) [inline src/components/Menu/Menu.vue]: please add params: {partyId: result.data.partyId}

File: src/components/Menu/Menu.vue [hunk 7]
```diff
@@ -1,66 +1,157 @@
...
+      return outputArray;
+    },
+    async createParty(e) {
       e.preventDefault();
 
       try {
-        const URL_PARTY = `${URL}/party`
-        const result = await this.$http.put(URL_PARTY)
-        await this.$router.push({name: 'CreateParty', params: {partyId: result.data.partyId}})
+        // const URL_PARTY = `${this.$env.VUE_APP_API_BASE_URL}/party`;
```
Comment (MaxenceMottard) [inline src/components/Menu/Menu.vue]: why comment

File: src/router/index.js [hunk 8]
```diff
@@ -23,7 +23,7 @@ const routes = [
             import("../components/Menu/JoinPartyStep2.vue")
     },
     {
-        path: "/create-party/:partyId",
+        path: "/create-party/:partyId", //TODO: pass partyId to optional
```
Comment (MaxenceMottard) [inline src/router/index.js]: remove this comment

**Review by MaxenceMottard — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (MrZyr0) [reply .github/workflows/deploy.yml]:
```suggestion
```

File: src/components/Menu/Menu.vue (hunk 5 above)
Comment (MaxenceMottard) [reply src/components/Menu/Menu.vue]: instead of process.env

File: public/sw.js (hunk 3 above)
Comment (MrZyr0) [reply public/sw.js:13]: No, we keep for next refactoring and have some security

File: public/sw.js (hunk 4 above)
Comment (MrZyr0) [reply public/sw.js]:
```suggestion
```

File: src/components/Menu/Menu.vue (hunk 6 above)
Comment (MaxenceMottard) [reply src/components/Menu/Menu.vue]: remove router-link

**Review by MaxenceMottard — CHANGES_REQUESTED** [review]

**Review by MaxenceMottard — APPROVED** [review]

**Pull request merged**